src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -104,7 +104,7 @@ private async Task<bool> TryInitializeAsync(
                 if (globalStatement != null)
                 {
                     InGlobalStatementContext = true;
-                    return true;
+                    return CanIntroduceVariable(textSpan.IsEmpty, cancellationToken);
```

Comment:
can you try just moving the call to `if (!CanIntroduceVariable(textSpan.IsEmpty, cancellationToken)` above this `if` block?

Reply:
(basically the code on line 113.  move that to line 100.

Reply:
@CyrusNajmabadi moved, my tests are passing, waiting for pipelines to run all tests :)